core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -553,12 +562,13 @@ private[spark] class ExecutorAllocationManager(
         }
 
         // If this is the last pending task, mark the scheduler queue as empty
-        stageIdToTaskIndices.getOrElseUpdate(stageId, new mutable.HashSet[Int]) += taskIndex
+        stageIdToTaskIndices.getOrElseUpdate(stageId, new mutable.HashSet[String]) += (taskIndex + "." + attemptId)
```

Comment:
yeah, I understand  what do you mean. But why I change the code is because, if a task is failed, a new one will append, this two tasks'  ```taskIndex``` are the same